howto-gallery/scripts/BlockEvent.ts

```diff
@@ -26,25 +28,32 @@ export function buttonPushEvent(log: (message: string, status?: number) => void,
   );
 
   mc.world.afterEvents.buttonPush.subscribe((buttonPushEvent: mc.ButtonPushAfterEvent) => {
-    let eventLoc = buttonPushEvent.block.location;
+    const eventLoc = buttonPushEvent.block.location;
 
     if (eventLoc.x === targetLocation.x && eventLoc.y === targetLocation.y + 1 && eventLoc.z === targetLocation.z) {
-      log("Button push event at tick " + mc.system.currentTick + " Power:" + buttonPushEvent.block.getRedstonePower());
+      log("Button push event at tick " + mc.system.currentTick);
     }
   });
 }
 
 /**
  * A simple lever activate event
  * @param {(message: string, status?: number) => void} log: Logger function. If status is positive, test is a success. If status is negative, test is a failure.
- * @param {mc.Location} location Location to center this sample code around.
+ * @param {mc.DimensionLocation} targetLocation Location to center this sample code around.
  * @see https://learn.microsoft.com/minecraft/creator/scriptapi/minecraft/server/leveractionaftereventsignal
  * @see https://learn.microsoft.com/minecraft/creator/scriptapi/minecraft/server/leveractionafterevent
  */
-export function leverActionEvent(log: (message: string, status?: number) => void, targetLocation: mc.Vector3) {
+export function leverActionEvent(
+  log: (message: string, status?: number) => void,
+  targetLocation: mc.DimensionLocation
+) {
   // set up a lever
-  let cobblestone = overworld.getBlock(targetLocation);
-  let lever = overworld.getBlock({ x: targetLocation.x, y: targetLocation.y + 1, z: targetLocation.z });
+  const cobblestone = targetLocation.dimension.getBlock(targetLocation);
+  const lever = targetLocation.dimension.getBlock({
+    x: targetLocation.x,
+    y: targetLocation.y + 1,
+    z: targetLocation.z,
+  });
```

Comment:
Same, it could be a usage sample for vanilla data (`Vector3Utils.add`)

Reply:
For basic operations like this I'd say it's probably not worth the dependency